api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -767,23 +767,53 @@ message Cluster {
 
   // HTTP protocol options that are applied only to upstream HTTP connections.
   // These options apply to all HTTP versions.
-  core.v3.UpstreamHttpProtocolOptions upstream_http_protocol_options = 46;
+  // This has been deprecated in favor of
+  // :ref:`upstream_http_protocol_options <envoy_api_field_extensions.filters.network.http_connection_manager.v3.HttpProtocolOptions.upstream_http_protocol_options>`
+  // in the :ref:`http_protocol_options <envoy_api_msg_extensions.filters.network.http_connection_manager.v3.HttpProtocolOptions>` message.
```

Comment:
Can you add to this issue (or even comments) some design rationale around why upstream options belong in HCM. I assume it has to do with upgrade or something like that.. Otherwise, it seems odd to have HCM (a network filter) specify options that belong on the cluster manager (and connection pool) side of Envoy's split downstream/upstream architecture.